src/opus_encoder.c

```diff
@@ -893,24 +893,28 @@ static opus_val32 compute_frame_energy(const opus_val16 *pcm, int frame_size, in
 
 /* Decides if DTX should be turned on (=1) or off (=0) */
 static int decide_dtx_mode(opus_int activity,            /* indicates if this frame contains speech/music */
-                           int *nb_no_activity_frames    /* number of consecutive frames with no activity */
+                           int *nb_no_activity_ms_Q2,    /* number of consecutive milliseconds with no activity, in Q2 */
+                           int frame_size_ms_Q2          /* number of miliseconds in this update, in Q2 */
                            )
 
 {
    if (!activity)
    {
-      /* The number of consecutive DTX frames should be within the allowed bounds */
-      (*nb_no_activity_frames)++;
-      if (*nb_no_activity_frames > NB_SPEECH_FRAMES_BEFORE_DTX)
+      /* The number of consecutive DTX frames should be within the allowed bounds. 
+      Note that the allowed bound is defined in the Silk headers and assumes 20 ms
+      frames. As this function can be called with any frame length, a conversion to
+      miliseconds is done before the comparisons. */
+      (*nb_no_activity_ms_Q2) += frame_size_ms_Q2;
+      if (*nb_no_activity_ms_Q2 > NB_SPEECH_FRAMES_BEFORE_DTX*20*2)
```

Comment:
I don't get the *2

Reply:
that counter, nb_no_activity_ms_Q2, is using a Q2 precission as we have to cover the case of 2.5 ms frames and we are using integers. So the *2 is needed for converting the NB_SPEECH_FRAMES_BEFORE_DTX to miliseconds in Q2.

Reply:
I see makes sense, I would probably add new defines with MS and Q2 in the names but might be overengineering it.

Reply:
also for old time sakes, what about test ;)

Reply:
I realize now that it is Q1, I will update the pull request. Yes, tests ... the ones that did not exist :) I have not looked yet how easy it is to add new ones.